metrics/api/src/test/java/io/helidon/metrics/api/MetricStoreTests.java

```diff
@@ -56,4 +60,70 @@ void testConflictingMetadata() {
+
+        NoOpMetricRegistry registry = NoOpMetricRegistry.create(MetricRegistry.Type.APPLICATION);
+
+        MetricStore store = MetricStore.create(REGISTRY_SETTINGS,
+                                               NoOpMetricRegistry.NO_OP_METRIC_FACTORIES,
+                                               null,
+                                               null,
+                                               MetricRegistry.Type.APPLICATION,
+                                               registry::toImpl);
+
+        Counter counter1 = store.getOrRegisterMetric(metadata, Counter.class, NO_TAGS);
+        Counter counter2 = store.getOrRegisterMetric(metadata, Counter.class, NO_TAGS);
+        assertThat("Counters with no tags", counter1, is(counter2));
+    }
+
+    @Test
+    void testSameNameSameTwoTags() {
+        Tag[] tags = {new Tag("foo", "1"), new Tag("bar", "1")};
+        Metadata metadata = Metadata.builder()
+                .withName("a")
+                .withType(MetricType.COUNTER)
+                .build();
+
+        NoOpMetricRegistry registry = NoOpMetricRegistry.create(MetricRegistry.Type.APPLICATION);
+
+        MetricStore store = MetricStore.create(REGISTRY_SETTINGS,
+                                               NoOpMetricRegistry.NO_OP_METRIC_FACTORIES,
+                                               null,
+                                               null,
+                                               MetricRegistry.Type.APPLICATION,
+                                               registry::toImpl);
+
+        Counter counter1 = store.getOrRegisterMetric(metadata, Counter.class, tags);
+        Counter counter2 = store.getOrRegisterMetric(metadata, Counter.class, tags);
+        assertThat("Counters with same two tags", counter1, is(counter2));
+    }
+
+    @Test
+    void testSameNameOverlappingButDifferentTags() {
+        Tag[] tags1 = {new Tag("foo", "1"), new Tag("bar", "1"), new Tag("baz", "1")};
+        Tag[] tags2 = {new Tag("foo", "1"), new Tag("bar", "1")};
+        Metadata metadata = Metadata.builder()
+                .withName("a")
+                .withType(MetricType.COUNTER)
+                .build();
+
+        NoOpMetricRegistry registry = NoOpMetricRegistry.create(MetricRegistry.Type.APPLICATION);
+
+        MetricStore store = MetricStore.create(REGISTRY_SETTINGS,
+                                               NoOpMetricRegistry.NO_OP_METRIC_FACTORIES,
+                                               null,
+                                               null,
+                                               MetricRegistry.Type.APPLICATION,
+                                               registry::toImpl);
+
+        Counter counter1 = store.getOrRegisterMetric(metadata, Counter.class, tags1);
+        Counter counter2 = store.getOrRegisterMetric(metadata, Counter.class, tags2);
+        assertThat("Counters with overlapping but different tags", counter1, not(is(counter2)));
```

Comment:
I would use either `is(not(` or just `not(` so the assertion reads like English sentence:
"assert that counter1 is not counter2" seems easier to understand than "assert that counter1 not is counter2"